src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -40,12 +54,44 @@ def consume(self, state):
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
 
-        for offset, message in consumer:
-            decoded = self.codec.decode(message.value)
-            state = validate_state(state, offset, decoded)
+        for offset, message in self.__primed(consumer, state):
+            state = validate_state(state, offset, message)
             # XXX: This is necessary because of a bug in protocol buffer oneof.
             state = type(state).FromString(state.SerializeToString())
-            yield state, offset, decoded
+            yield state, offset, message
+
+    def __primed(self, consumer, state, attempts=5):
```

Comment:
That was probably the wrong phrase, yeah.  What I was trying to say was the code to do this on the `KafkaStream` class is already written, and tested and is easy to extract out once there is a need for a common abstraction. Doing the extraction work right now isn't as high a priority as getting the 0.9 milestone done so we can run it against our production database (not this is not *in* production).

In light of everything else and the other comments it's likely it will get abstracted out here anyways so it doesn't matter that much.